src/Components/books/book.jsx

```diff
@@ -0,0 +1,588 @@
+import React, { useEffect, useState } from "react";
+import HocContainer from "../HocContainer";
+import "../../styles/books.css";
+import Button from "../Button";
+import Modal from "../modal";
+import Table from "../Table";
+import SearchBox from "../searchBox";
+import {
+  fetch_delete,
+  fetch_get,
+  fetch_put,
+  fetch_post,
+} from "../../api/apiManager";
+import Toast from "../Toast";
+import Dropdown from "../Dropdown";
+import { useNavigate } from "react-router-dom";
+import Loader from "../loader";
+import ActionIcon from "../../Assets/Icons/more.png";
+
+const Books = () => {
+  const navigate = useNavigate();
+  const [toastMessage, setToastMessage] = useState(null);
+  const [toastType, setToastType] = useState("success");
+  const [books, setBooks] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [editingBookId, setEditingBookId] = useState(null);
+  const [bookTitle, setBookTitle] = useState("");
+  const [quantity, setQuantity] = useState(null);
+  const [bookAuthor, setBookAuthor] = useState("");
+  const [categoryForBook, setCategoryForBook] = useState("");
+  const [isDeleteConfirmation, setIsDeleteConfirmation] = useState(false);
+  const [bookToDelete, setBookToDelete] = useState(null);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [totalPages, setTotalPages] = useState(0);
+  const [selectedBook, setSelectedBook] = useState(null);
+  const [mobileNumber, setMobileNumber] = useState("");
+  const [issuanceType, setIssuanceType] = useState("");
+  const [status, setStatus] = useState("");
+  const [issueError, setIssueError] = useState(null);
+  const [returnedAt, setReturnedAt] = useState(null);
+  const [searchQuery, setSearchQuery] = useState("");
+  const [historyData, setHistoryData] = useState([]);
+  const [isHistoryModalOpen, setIsHistoryModalOpen] = useState(false);
+  const [debouncedSearchQuery, setDebouncedSearchQuery] = useState("");
+  const [userName, setUserName] = useState("");
+  
+  const fetchBooksAndCategories = async () => {
+    setLoading(true); 
+    try {
+      const [booksResponse, categoriesResponse] = await Promise.all([
+        fetch_get("/api/books/list", {
+          page: currentPage,
+          size: 5,
+          search: debouncedSearchQuery,
+        }),
+        fetch_get("/api/categories"),
+      ]);
+
+      setBooks(booksResponse.data.content);
+      setTotalPages(booksResponse.data.totalPages);
+      setCategories(categoriesResponse.data);
+      setError(null);
+    } catch (err) {
+      setError("Failed to load books, try again later!");
```

Comment:
please maintain a constants file to store these messages
or please display the message which are in API success/error response.